apps/wallet/src/ui/app/pages/home/tokens/ReceiveTokensDialog.tsx

```diff
@@ -12,10 +20,39 @@ interface ReceiveTokensDialogProps {
 }
 
 export function ReceiveTokensDialog({ address, open, setOpen }: ReceiveTokensDialogProps) {
+    const activeAccount = useActiveAccount();
+    const { connectToLedger, iotaLedgerClient } = useIotaLedgerClient();
+
     const onCopy = useCopyToClipboard(address, {
         copySuccessMessage: 'Address copied',
     });
 
+    const isLedger = isLedgerAccountSerializedUI(activeAccount as LedgerAccountSerializedUI);
+
+    const onVerifyAddress = useCallback(() => {
+        if (isLedger && activeAccount) {
+            (async () => {
+                try {
+                    let ledgerClient = iotaLedgerClient;
+                    if (!ledgerClient) {
+                        ledgerClient = await connectToLedger(true);
+                    }
+
+                    const derivationPath = (activeAccount as LedgerAccountSerializedUI)
+                        .derivationPath;
+
+                    if (derivationPath) {
+                        toast.success('Please, confirm the address on your Ledger device.');
+                        await ledgerClient.getPublicKey(derivationPath, true);
+                        toast.success('Address verification successful!');
+                    }
+                } catch {
+                    toast.error('Address verification failed!');
+                }
+            })();
+        }
+    }, [isLedger, activeAccount, iotaLedgerClient, connectToLedger]);
```

Comment:
Just some code improvements suggestions
```suggestion
    const onVerifyAddress = useCallback(async () => {
        if(!isLedger || !activeAccount){
            return
        }

        if(!isLedgerAccountSerializedUI(activeAccount)){
            return
        }

        try {
            let ledgerClient = iotaLedgerClient;
            if (!ledgerClient) {
                ledgerClient = await connectToLedger(true);
            }
            
            toast.success('Please, confirm the address on your Ledger device.');
            await ledgerClient.getPublicKey(activeAccount.derivationPath, true);
            toast.success('Address verification successful!');
        } catch {
            toast.error('Address verification failed!');
        }
    }, [isLedger, activeAccount, iotaLedgerClient, connectToLedger]);
```

Reply:
Applied suggestion in [53117d7](https://github.com/iotaledger/iota/pull/4151/commits/53117d7dda45cd88443db061758015e7df17db20)